app/src/main/java/com/example/filemanager/AllFilesFragment.kt

```diff
@@ -59,4 +71,65 @@ class AllFilesFragment : Fragment(), ItemClickListener {
     override fun onItemClickListener() {
         Toast.makeText(context, "Item clicked", Toast.LENGTH_SHORT).show()
     }
+
+    private fun showPermissionDialog() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+            try {
+                val intent = Intent(Settings.ACTION_MANAGE_ALL_FILES_ACCESS_PERMISSION)
+                intent.addCategory("android.intent.category.DEFAULT")
+                intent.data = Uri.parse(String.format("package:%s", arrayOf<Any>(applicationContext().packageName)))
+                activity?.startActivityForResult(intent, 2000)
```

Comment:
Could you please try in a different branch maybe to just grant `WRITE_EXTERNAL_STORAGE` permission and remove the `READ_EXTERNAL_STORAGE` and `MANAGE_EXTERNAL_STORAGE` and also remove this part with enabling it from settings `ACTION_MANAGE_ALL_FILES_ACCESS_PERMISSION`. And see if it works for you, because it works for me, however the correct way would be to use the `MANAGE_EXTERNAL_STORAGE` for versions 30+. But it works for me without it which is very strange.

Reply:
So, I also tried to delete MANAGE_EXTERNAL_STORAGE and READ_EXTERNAL_STORAGE permissions, and it works on all devices but on android 11 it doesn't show all files